test/plugins/WebOfTrust/util/TickerDelayedBackgroundJobTest.java

```diff
@@ -0,0 +1,393 @@
+    }
+
+    @Test
+    public void testTerminate() throws Exception {
+        // Test immediate termination on IDLE
+        TickerDelayedBackgroundJob job1 = newJob(50, 20, "terminate1");
+        assertEquals(JobState.IDLE, job1.getState());
+        assertFalse(job1.isTerminated());
+        job1.terminate();
+        assertEquals(JobState.TERMINATED, job1.getState());
+        assertTrue(job1.isTerminated());
+        assertFalse(wasInterrupted.get());
+
+        // Test immediate termination on WAITING
+        TickerDelayedBackgroundJob job2 = newJob(50, 20, "terminate2");
+        assertEquals(JobState.IDLE, job2.getState());
+        assertFalse(job2.isTerminated());
+        job2.triggerExecution();
+        assertEquals(JobState.WAITING, job2.getState());
+        assertFalse(job2.isTerminated());
+        job2.terminate();
+        assertEquals(JobState.TERMINATED, job2.getState());
+        assertTrue(job2.isTerminated());
+        assertFalse(wasInterrupted.get());
+
+        // Test interrupting termination on RUNNING
+        TickerDelayedBackgroundJob job3 = newJob(50, 20, "terminate3");
+        assertEquals(JobState.IDLE, job3.getState());
+        assertFalse(job3.isTerminated());
+        job3.triggerExecution(0);
+        Thread.sleep(20);
+        assertEquals(JobState.RUNNING, job3.getState());
+        // Synchronize here to avoid the race condition where the thread has terminated before we
+        // have a chance to inspect its TERMINATING state.
+        synchronized(job3) {
+            job3.terminate();
+            assertEquals(JobState.TERMINATING, job3.getState());
+            assertFalse(job3.isTerminated());
+        }
+        Thread.sleep(20);
+        assertEquals(JobState.TERMINATED, job3.getState());
+        assertTrue(job3.isTerminated());
+        assertTrue(wasInterrupted.get());
+        // Reset interrupted flag, otherwise our @After {@link #checkCanaries()} will throw.
+        wasInterrupted.set(false);
+    }
+
+    @Test
+    public void testWaitForTermination() throws Exception {
+        long begin, end;
+        // Test that the timeout is obeyed within reasonable limits (at most 10 ms too much).
+        DelayedBackgroundJob job1 = newJob(0, 50, "wait1");
+        for (int i = 0; i < 10; i++) {
+            long timeout = 10 * i;
+            begin = System.currentTimeMillis();
+            job1.waitForTermination(timeout);
+            end = System.currentTimeMillis();
+            long waited = end - begin;
+            assertTrue(waited >= timeout);
+            assertTrue(waited <= timeout + 10);
+        }
+
+        // Test that terminated jobs return reasonably immediately.
+        DelayedBackgroundJob job2 = newJob(0, 50, "wait2");
+        job2.terminate();
+        begin = System.currentTimeMillis();
+        job2.waitForTermination(1000);
+        end = System.currentTimeMillis();
+        assertTrue(end - begin < 2);
+
+        // Test termination from job and notify
```

Comment:
Notice: You don't have to fix the existing questions anymore, just tell me the answer, I will fix them.